Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -124,66 +124,178 @@ namespace AzToolsFramework
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Action Manager - Could not set name of category \"%s\" as no action with that identifier was registered.",
+                actionIdentifier.c_str()));
+        }
+
+        actionIterator->second.SetCategory(category);
+        return AZ::Success();
+    }
+
+    ActionManagerGetterResult ActionManager::GetActionIconPath(const AZStd::string& actionIdentifier)
+    {
+        auto actionIterator = m_actions.find(actionIdentifier);
+        if (actionIterator == m_actions.end())
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Action Manager - Could not get icon path of action \"%s\" as no action with that identifier was registered.",
+                actionIdentifier.c_str()));
+        }
+
+        return AZ::Success(actionIterator->second.GetIconPath());
+    }
+
+    ActionManagerOperationResult ActionManager::SetActionIconPath(const AZStd::string& actionIdentifier, const AZStd::string& iconPath)
+    {
+        auto actionIterator = m_actions.find(actionIdentifier);
+        if (actionIterator == m_actions.end())
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Action Manager - Could not set icon path of action \"%s\" as no action with that identifier was registered.",
+                actionIdentifier.c_str()));
+        }
+        
+        actionIterator->second.SetIconPath(iconPath);
+        return AZ::Success();
+    }
 
     ActionManagerOperationResult ActionManager::TriggerAction(const AZStd::string& actionIdentifier)
     {
-        if (!m_actions.contains(actionIdentifier))
+        auto actionIterator = m_actions.find(actionIdentifier);
+        if (actionIterator == m_actions.end())
         {
             return AZ::Failure(AZStd::string::format(
                 "Action Manager - Could not trigger action \"%s\" as no action with that identifier was registered.",
                 actionIdentifier.c_str()));
         }
 
-        m_actions[actionIdentifier].GetAction()->trigger();
+        actionIterator->second.GetAction()->trigger();
         return AZ::Success();
     }
 
     QAction* ActionManager::GetAction(const AZStd::string& actionIdentifier)
     {
-        if (!m_actions.contains(actionIdentifier))
+        auto actionIterator = m_actions.find(actionIdentifier);
+        if (actionIterator == m_actions.end())
         {
             return nullptr;
         }
 
-        return m_actions[actionIdentifier].GetAction();
+        return actionIterator->second.GetAction();
     }
 
     const QAction* ActionManager::GetActionConst(const AZStd::string& actionIdentifier)
     {
-        if (!m_actions.contains(actionIdentifier))
+        auto actionIterator = m_actions.find(actionIdentifier);
+        if (actionIterator == m_actions.end())
         {
             return nullptr;
         }
-
-        return m_actions[actionIdentifier].GetAction();
+        
```

Comment:
Nit: Looks like some extra whitespace here

Reply:
Whoops, will fix in follow-up!